src/HookHandlers.php

```diff
@@ -13,4 +19,22 @@ public static function onScribuntoExternalLibraries( string $engine, array &$ext
 		$extraLibraries['SPARQL'] = SparqlLua::class;
 	}
 
+	public static function onParserTestGlobals( array &$globals ): void {
+		MediaWikiServices::getInstance()->resetServiceForTesting( 'HttpRequestFactory' );
+		MediaWikiServices::getInstance()->redefineService(
+			'HttpRequestFactory',
+			// The below function is copied from ServiceWiring.php. Not sure how to access the original function.
+			static function ( MediaWikiServices $services ): HttpRequestFactory {
+				return new HttpRequestFactory(
+					new ServiceOptions(
+						HttpRequestFactory::CONSTRUCTOR_OPTIONS,
+						$services->getMainConfig()
+					),
+					LoggerFactory::getInstance( 'http' ),
+					Telemetry::getInstance()
+				);
```

Comment:
Copying this is one big meh. The reason we do our own service construction is to avoid dealing with the MediaWiki service container and its magical spaghetti.